hilti/toolchain/src/compiler/validator.cc

```diff
@@ -631,6 +631,11 @@ struct VisitorPost : visitor::PreOrder, public validator::VisitorMixIn {
             error(fmt("integer type's width must be one of 8/16/32/64, but is %d", n->width()), n);
     }
 
+    void operator()(type::Set* n) final {
+        if ( auto rc = isSortable(n->elementType()); ! rc )
+            error(fmt("type cannot be used as element type for sets (because %s)", rc.error()), n);
```

Comment:
We trigger the change in the baseline of `hilti.types.set.ctor-fail` since for unresolved types `n->elementType()` is `type::Unknown` which is not sortable; the diagnostic is however not interesting to users since original error is elsewhere and lead to us not being able to compute an element type. The fix is probably to only report the error if we have a valid type.

```suggestion
        if ( ! n->elementType()->type()->isA<type::Unknown>() ) { // unknown will be reported elsewhere
            if ( auto rc = isSortable(n->elementType()); ! rc )
                error(fmt("type cannot be used as element type for sets (because %s)", rc.error()), n);
        }
```

While at it we should make a similar adjustment for the check on `type::Map` above.

Reply:
I mainly just assumed that we'd want to catch unknown if it gets there without a ctor, but now I don't think I can make a case where that happens. So - agreed :)

if it does later it should be a separate error anyway